cli/root_test.go

```diff
@@ -16,8 +20,78 @@ import (
 	"github.com/coder/coder/cli/cliflag"
 	"github.com/coder/coder/cli/clitest"
 	"github.com/coder/coder/codersdk"
+	"github.com/coder/coder/testutil"
 )
 
+var updateGoldenFiles = flag.Bool("update", false, "update .golden files")
+
+//nolint:tparallel,paralleltest // These test sets env vars.
+func TestCommandHelp(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name string
+		cmd  []string
+		env  map[string]string
+	}{
+		{
+			name: "coder --help",
+			cmd:  []string{"--help"},
+			env: map[string]string{
+				"CODER_CONFIG_DIR": "/tmp/coder-cli-test-config",
+			},
+		},
+		{
+			name: "coder server --help",
+			cmd:  []string{"server", "--help"},
+			env: map[string]string{
+				"CODER_CONFIG_DIR":      "/tmp/coder-cli-test-config",
+				"CODER_CACHE_DIRECTORY": "/tmp/coder-cli-test-cache",
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			// Unset all CODER_ environment variables for a clean slate.
```

Comment:
I'm wondering if you shouldn't remove all environment variables, like TTY-related, `NO_COLOR`, etc.

Reply:
I was being cautious by only overriding the CODER ones, but I'm guessing it won't hurt 👍🏻. Currently (to my knowledge) the only non-coder env that has an effect on the application is `CACHE_DIRECTORY` (for systemd compat).

Reply:
Dang, looks like it broke Windows, I'll just revert the change and we can fix any issues if they surface in the future.

Reply:
Interesting. Maybe there is a minimal set of env vars that are required by Windows.